capa/features/extractors/ida/helpers.py

```diff
@@ -331,3 +331,28 @@ def is_basic_block_tight_loop(bb):
             if ref == bb.start_ea:
                 return True
     return False
+
+
+def find_data_reference_from_insn_helper(ea, max_depth=10):
+    """ recursive search for data reference, if exists, from instruction by resolving nested pointers, if exist """
```

Comment:
Personally, I find the iterative approach in `derefs` (`capa\features\extractors\viv\insn.py`) easier to follow and hence to maintain.

Reply:
I've pushed an iterative solution similar to `derefs`.